### DISCUSSION TITLE
TeamViewer winget uninstall failing

### DISCUSSION THREAD

**Discussion by dlundahl91** [Romanitho/Winget-Install#88] [Q&A] [answered]
Hey team!

When I install Teamviewer.Teamviewer with the winget-install, it installs just fine. (under system context, as it gets installed in Program Files)
However, the uninstallation does not work, it gives this result in the install logs:

```
08:11:31 - ###   10/7/2023 - NEW UNINSTALL REQUEST   ###
08:11:31 - Using following Winget Cmd: C:\Program Files\WindowsApps\microsoft.desktopappinstaller_1.21.2721.0_x64__8wekyb3d8bbwe\winget.exe

08:11:31 - Start TeamViewer.TeamViewer processing...
08:11:32 - -> Uninstalling TeamViewer.TeamViewer...
08:11:32 - -> Running: "C:\Program Files\WindowsApps\microsoft.desktopappinstaller_1.21.2721.0_x64__8wekyb3d8bbwe\winget.exe" uninstall --id TeamViewer.TeamViewer -e --accept-source-agreements -h

   - 
                                                                                                                        

  ███▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  1024 KB / 8.07 MB
  ███████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  2.00 MB / 8.07 MB
  ███████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  3.00 MB / 8.07 MB
  ██████████████▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒  4.00 MB / 8.07 MB
  ██████████████████▒▒▒▒▒▒▒▒▒▒▒▒  5.00 MB / 8.07 MB
  ██████████████████████▒▒▒▒▒▒▒▒  6.00 MB / 8.07 MB
  █████████████████████████▒▒▒▒▒  7.00 MB / 8.07 MB
  █████████████████████████████▒  8.00 MB / 8.07 MB
  ██████████████████████████████  8.07 MB / 8.07 MB
                                                                                                                        
Found TeamViewer [TeamViewer.TeamViewer]
Starting package uninstall...

   - 
                                                                                                                        
Successfully uninstalled
08:11:34 - -> TeamViewer.TeamViewer uninstall failed!
08:11:34 - TeamViewer.TeamViewer processing finished!

08:11:35 - ###   END REQUEST   ###
```

Any ideas? I'd love some help/input!

**Comment by dlundahl91** [comment]
Hey team!

So I decided to try and uninstall it manually by using
`winget uninstall Teamviewer.Teamviewer`

But after using this, I get a popup, looking like this:
<img width="375" alt="popup1" src="https://github.com/Romanitho/Winget-Install/assets/144660401/9a57be11-344f-46af-9ae4-15294e1b9461">

After popup, I tried to write in:
`winget uninstall --help`

and learned it is possible for me to use this command instead
`winget uninstall Teamviewer.Teamviewer --silent`

However, it still comes up with the popup. Could that be why the winget uninstall fails?

Do you have any idea how I may fix this uninstall issue?

Thanks so much in advance guys!

**Reply by KnifMelti** [reply]
Winget-Install is already running a silent uninstall:
`Running: "C:\Program Files\WindowsApps\microsoft.desktopappinstaller_1.21.2721.0_x64__8wekyb3d8bbwe\winget.exe" uninstall --id TeamViewer.TeamViewer -e --accept-source-agreements -h`

-h = --silent `-h,--silent                 Request silent installation`

As @Romanitho already points out:
https://github.com/Romanitho/Winget-Install#uninstall

**Reply by Romanitho** [reply]
Exactly. The Winget uninstall part is not very well managed in my opinion. I often use native uninstallers.

**Comment by dlundahl91** [comment]
As a temporary solution, I have found this intune uninstall command to work:
`"C:\Program Files\TeamViewer\uninstall.exe" /S`

I would however greatly appreciate input when you guys have time. Look forward to hearing from you when the business days shine again. B-)

**Reply by KnifMelti** [reply, marked as answer]
That's a manifest issue, it's an **InstallerType: nullsoft** with no override parameters:
https://github.com/microsoft/winget-pkgs/blob/master/manifests/t/TeamViewer/TeamViewer/15.46.5/TeamViewer.TeamViewer.installer.yaml
Your own workaround is one way to solve it, **mods** is another.
https://github.com/Romanitho/Winget-Install#custom-mods
Rename to .ps1: [TeamViewer.TeamViewer-preuninstall.txt](https://github.com/Romanitho/Winget-Install/files/12839013/TeamViewer.TeamViewer-preuninstall.txt)
